### PR TITLE
feat(react): improve storybook schematics

### PR DESCRIPTION
**Pull request by juristr**
## Current Behavior (This is the behavior we have today, before the PR is merged)

React schematics only have basic storybook configuration support.

## Expected Behavior (This is the new behavior we can expect after the PR is merged)

We should have a similar experience as the Angular Nx storybook schematics provide. Like generating storybooks for the components in a library, do that right upfront when generating a new component library etc..

### CODE REVIEW

**Comment by jaysoo** [comment]
(Added to the review, but will comment here for visibility)

We should look for the a default export from the component file, and then check that the default export is either:

1. A function that returns JSX in its return type; OR
2. A class that has a `render()` method that returns JSX.

Note that all of the following are valid:

```
// 1) export default function
export default function Button() {
  return <button>...</button>;
}

// 2) function declaration + default export
function Button() {
  return <button>...</button>;
}
export default Button;

// 3) arrow function + default export
const Button = () => <button>...</button>;
export default Button

// 4) export default class
export default class Button extends React.Component {
  render() {
    return <button>...</button>;
  }
}

// 5) class declaration + default export 
class Button extends React.Component {
  render() {
    return <button>...</button>;
  }
}
export default Button;
```

---

There is a lot more guesswork since React doesn't provide any module definitions like Angular, but going with default export is a safe assumption since almost all projects do it this way.

**Comment by juristr** [comment]
> We should look for the a default export from the component file, and then check that the default export is either: [...]

@jaysoo good point. Yeah indeed that was kind of an open question for me as well. Thx for the hints.

**Comment by juristr** [comment]
So this PR contains quite some things (as I discovered some more issues moving along).

🆕 schematic for generating a story for a single component
🆕 schematic for generating stories for components of an entire library
✅ update storybook configuration to include the option to generate stories automatically
✅ improved support for JS-based stories 
✅ fixed proper detection of default component export in ast-utils
✅ fixed issue with `@babel/preset-typescript` missing & leading to a runtime issue with storybook
✅ fixed Cypress setup when JS only specs are being generated

I found some more issues with our Storybook & React setup. I'll provide a new PR to fix those.

**Comment by Hotell** [comment]
ha just wanted to report -> `fixed issue with @babel/preset-typescript missing & leading to a runtime issue with storybook` :D 

So we'll have to wait to get this merged or you gonna create new PR's which fixes those issues ?

awesome work ! thx

**Comment by Hotell** [comment]
> (Added to the review, but will comment here for visibility)
> 
> We should look for the a default export from the component file, and then check that the default export is either:
> 
> 1. A function that returns JSX in its return type; OR
> 2. A class that has a `render()` method that returns JSX.
> 
> Note that all of the following are valid:
> 
> ```
> // 1) export default function
> export default function Button() {
>   return <button>...</button>;
> }
> 
> // 2) function declaration + default export
> function Button() {
>   return <button>...</button>;
> }
> export default Button;
> 
> // 3) arrow function + default export
> const Button = () => <button>...</button>;
> export default Button
> 
> // 4) export default class
> export default class Button extends React.Component {
>   render() {
>     return <button>...</button>;
>   }
> }
> 
> // 5) class declaration + default export 
> class Button extends React.Component {
>   render() {
>     return <button>...</button>;
>   }
> }
> export default Button;
> ```
> 
> There is a lot more guesswork since React doesn't provide any module definitions like Angular, but going with default export is a safe assumption since almost all projects do it this way.

**I'd add:**
- `class Button extends React.PureComponent` (if schematic will do evaluation based on token type - super class)

> but going with default export is a safe assumption since almost all projects do it this way.

That depends. for instance I'm never using `default exports` for various legit reasons, my team neither (I'd accommodate both default and named exports - with defaults having priority /if both exports are present with same token name/ - because, yes, I do agree/it's a fact, that "unfortunately" most of react ecosystem/community is using default exports )

**Comment by juristr** [comment]
> So we'll have to wait to get this merged or you gonna create new PR's which fixes those issues ?

@Hotell it'll get merged in & fixed with this one :)

> class Button extends React.PureComponent (if schematic will do evaluation based on token type - super class)

Thanks for this. Gonna check whether this works with my PR & in case update

File: packages/react/src/schematics/stories/stories.ts [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+    return chain(
+      componentPaths.map(componentPath => {
+        const relativeCmpDir = componentPath.replace(
+          join('/', projectSrcRoot, '/'),
+          ''
+        );
+
+        return chain([
+          schematic<CreateComponentStoriesFileSchema>('component-story', {
+            componentPath: relativeCmpDir,
```
Comment (jaysoo) [inline packages/react/src/schematics/stories/stories.ts:48]: Would there be any problems with a `.js` file that doesn't contain any React components?

**Review by jaysoo — APPROVED** [review]

File: packages/react/src/schematics/stories/stories.ts (hunk 1 above)
Comment (juristr) [reply packages/react/src/schematics/stories/stories.ts:48]: fixed

**Comment by juristr** [comment]
@Hotell Fixed `PureComponent`, so that should work now. 

> That depends. for instance I'm never using default exports for various legit reasons, my team neither (I'd accommodate both default and named exports - with defaults having priority /if both exports are present with same token name/ - because, yes, I do agree/it's a fact, that "unfortunately" most of react ecosystem/community is using default exports )

@Hotell I'll submit a follow-up PR for also searching non-default exported components as I really want to get this PR in. It already got huge 😅

**Comment by Hotell** [comment]
Sure thing! This is so awesome 👩‍🎤

**Pull request merged**